service/lib/dinstaller/luks_question.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dinstaller/question"
+
+module DInstaller
+  # This class represent a question to ask whether to activate a LUKS device
+  class LuksQuestion < Question
+    # Constructor
+    #
+    # @param device [String] name of the device to be activated (e.g., "/dev/sda1")
+    # @param label [String, nil] label of the device
+    # @param size [String, nil] size of the device (e.g., "5 GiB")
+    def initialize(device, label: nil, size: nil)
+      @device = device
+      @label = label
+      @size = size
+
+      super(generate_text, options: [:skip, :decrypt])
+    end
+
+    # Password to activate the LUKS device
+    #
+    # @return [String, nil]
```

Comment:
please document what means nil here. Also what means empty password. ( I have suspicious that something should mean to not activate luks, but maybe it is in answer and not in password.

Reply:
Yes, it is in answer. The `nil` password means nothing has been provided yet. Setting an empty password should be prevented by the client, I guess.